### PR TITLE
chore(#8700): clean up and organize e2e tests folders /service-worker, /sms, /targets, /tasks

### PR DESCRIPTION
**Pull request by tatilepizs**
# Description

Clean up and organize the following e2e tests:

- [service-worker](https://github.com/medic/cht-core/tree/master/tests/e2e/default/service-worker)
  - [x] service-worker.wdio-spec.js

- [sms](https://github.com/medic/cht-core/tree/master/tests/e2e/default/sms)
  - [x] africas-talking.wdio-spec.js
  - [x] export.wdio-spec.js
  - [x] gateway.wdio-spec.js
  - [x] link-to-patient.wdio-spec.js
  - [x] messages-sender-data.wdio-spec.js
  - [x] rapidpro.wdio-spec.js
  - [x] send-message.wdio-spec.js

- [targets](https://github.com/medic/cht-core/tree/master/tests/e2e/default/targets)
  - [x] target-aggregates.wdio-spec.js

- [tasks](https://github.com/medic/cht-core/tree/master/tests/e2e/default/tasks)
  - [x] due-dates.wdio-spec.js
  - [x] tasks-breadcrumbs.wdio-spec.js
  - [x] tasks-group.wdio-spec.js
  - [x] tasks.wdio-spec.js

- [page-objects](https://github.com/medic/cht-core/tree/master/tests/page-objects/default)
  - [x] messages.wdio.page.js
  - [x] target-aggregates.wdio.page.js
  - [x] tasks.wdio.page.js

#8700 

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# Compose URLs
If Build CI hasn't passed, these may 404:

* [Core](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8700-clean-up-and-organize-folders/docker-compose/cht-core.yml)
* [CouchDB Single](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8700-clean-up-and-organize-folders/docker-compose/cht-couchdb.yml)
* [CouchDB Cluster](https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:8700-clean-up-and-organize-folders/docker-compose/cht-couchdb-clustered.yml)

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/e2e/default/sms/rapidpro.wdio-spec.js
Comment (lorerod) [inline tests/e2e/default/sms/rapidpro.wdio-spec.js:53]: This is duplicated in afterEach hook.

File: tests/e2e/default/sms/rapidpro.wdio-spec.js
Comment (lorerod) [inline tests/e2e/default/sms/rapidpro.wdio-spec.js:297]: If it is updating settings in the beforeEach hook, can you please check if adding a revertSetting in the afterEach hook or maybe the afterHook is necessary? 
The updateSettings also have an option to revertSettings before updating.

File: tests/e2e/default/service-worker/service-worker.wdio-spec.js
Comment (lorerod) [inline tests/e2e/default/service-worker/service-worker.wdio-spec.js:102]: Is this necessary in afterEach? L198 is also doing a revertSettings, and I wonder if every test needs to revert.

**Review by lorerod — CHANGES_REQUESTED** [review]
Hi @tatilepizs, thank you for these improvements. I left some suggestions inline.

File: tests/e2e/default/tasks/tasks-group.wdio-spec.js
Comment (m5r) [inline tests/e2e/default/tasks/tasks-group.wdio-spec.js:16]: FYI for future usage, there is the static `Date.now()` method to do this more simply
```ts
new Date().getTime() === Date.now() // true
```

No need to change anything here, this is totally valid :green_circle:

**Review by m5r — APPROVED** [review]
This looks great! I left a few comments but this is good to go (pending Lorena's approval ofc)

File: tests/e2e/default/tasks/tasks-breadcrumbs.wdio-spec.js
Comment (tatilepizs) [reply tests/e2e/default/tasks/tasks-breadcrumbs.wdio-spec.js:25]: Because for some reason if a use the `personFactory` for those two (`chwContact` and `supervisorContact`) the tasks are in a different order and one more task is created, and I didn't want to change the expected values.

This is how it looks with the objects as they are:
image

And this is how it looks if I use factories:

```.js
  const chwContact = personFactory.build({
    _id: 'fixture:user:user1',
    name: 'chw',
    phone: '+12068881234',
    type: 'person',
    place: healthCenter1._id,
    parent: healthCenter1
  });
  const supervisorContact = personFactory.build({
    _id: 'fixture:user:user2',
    name: 'supervisor',
    phone: '+12068881235',
    type: 'person',
    place: districtHospital._id,
    parent: districtHospital
  });
```

image

What do you think @lorerod ? should I change the expected values in the assert?

File: tests/e2e/default/sms/rapidpro.wdio-spec.js
Comment (tatilepizs) [reply tests/e2e/default/sms/rapidpro.wdio-spec.js:297]: I added the `revertSettings` in the `afterEach` from the first `describe`, in that way it will be valid for all the `it`, and also removed this `revertDb` call because it is being duplicated with the `afterEach` from the first describe. Thank you Lore!

File: tests/e2e/default/service-worker/service-worker.wdio-spec.js
Comment (tatilepizs) [reply tests/e2e/default/service-worker/service-worker.wdio-spec.js:102]: good catch! ⭐

File: tests/e2e/default/tasks/tasks-breadcrumbs.wdio-spec.js
Comment (lorerod) [reply tests/e2e/default/tasks/tasks-breadcrumbs.wdio-spec.js:25]: I can only think of is that the person factory sets `patiend_id` to a value, but why not generate tasks also for the supervisor? The parent place may have something to do with it. Do you mind setting `patien_id` to a null value to try?

**Review by lorerod — APPROVED** [review]
Thank you for implementing the suggested feedback. This LGTM!

File: tests/e2e/default/tasks/tasks-breadcrumbs.wdio-spec.js
Comment (tatilepizs) [reply tests/e2e/default/tasks/tasks-breadcrumbs.wdio-spec.js:25]: Great question, I don't know what is different either between the `cht` and the `supervisor`, but adding the null value had the same result
image

Comment (tatilepizs) [reply tests/e2e/default/tasks/tasks-breadcrumbs.wdio-spec.js:25]: Ok, I understand now

the important attribute was the `reported_date`, in the config file, [here](https://github.com/medic/cht-core/blob/master/tests/e2e/default/tasks/config/tasks-breadcrumbs-config.js#L8) is where it evaluates the date. So if I set the `reported_date` as null it will work.

And the creation of the `chw` or `supervisor` was due to the user that was logged in.

Thank you Lore! ⭐

**Pull request merged**
